src/Analyzer/Passes/LogicalExpressionOptimizerPass.cpp

```diff
@@ -486,7 +486,7 @@ class LogicalExpressionOptimizerVisitor : public InDepthQueryTreeVisitorWithCont
             /// first we create tuple from RHS of equals functions
             for (const auto & equals : equals_functions)
             {
-                is_any_nullable |= equals->getResultType()->isNullable();
+                is_any_nullable |= removeLowCardinality(equals->getResultType())->isNullable();
```

Comment:
Can `in_function->getResultType()`tat we check below  also be `LowCardinality(Nullable)` ?

Reply:
hmm not sure, but if it wasn't `LowCardinality(Nullable)` in this case I don't know when it can be